api-server/web-server/src/api/v1.rs

```diff
@@ -68,115 +81,242 @@ pub fn routes() -> Router<APIServerWebServerState> {
 //
 
 #[allow(clippy::unused_async)]
-pub async fn block(
-    Path(_block_id): Path<Id<String>>,
+pub async fn block<T: ApiServerStorage>(
+    Path(block_id): Path<String>,
+    State(state): State<APIServerWebServerState<Arc<T>>>,
 ) -> Result<impl IntoResponse, APIServerWebServerError> {
-    // TODO replace mock with database calls
-
-    let mut rng = make_true_rng();
-
-    Ok(Json(json!({
-        "previous_block_id": Id::<Block>::new(H256::random_using(&mut rng)),
-        "height": BlockHeight::from(rng.gen_range(1..1_000_000)),
-        "timestamp": rng.gen_range(1..1_000_000_000),
-        "merkle_root": H256::random_using(&mut rng),
-        "reward": (0..rng.gen_range(1..5)).map(|_| { json!({
-                "destination": H256::random_using(&mut rng),
-                "amount": rng.gen_range(1..100_000_000),
-            })
-        }).collect::<Vec<_>>(),
-        "transaction_ids": (0..rng.gen_range(1..100)).map(|_| {
-            Id::<Transaction>::new(H256::random_using(&mut rng)
-        )}).collect::<Vec<_>>(),
-    })))
+    let block = {
+        let block_id: Id<Block> = H256::from_str(&block_id)
+            .map_err(|_| {
+                APIServerWebServerError::ClientError(APIServerWebServerClientError::BadRequest)
+            })?
+            .into();
+
+        state
+            .db
+            .transaction_ro()
+            .map_err(|_| {
+                APIServerWebServerError::ServerError(
+                    APIServerWebServerServerError::InternalServerError,
+                )
+            })?
```

Comment:
Could you define a constant or something for the error? I don't like how a mere conversion of error type takes up 5 lines while the actual "substantive" code is a fraction of that.

Reply:
This is replicated many times over.

Reply:
Yep. There's currently not that many error types, but once it's fully fleshed out, I'll replace the errors with real constants.